py/desitarget/lyazcat.py

```diff
@@ -0,0 +1,463 @@
+    data_out = fits.HDUList([prim_hdu, data_hdu])
+    
+    data_out.writeto(full_outputname)
+
+    return full_outputname
+
+
+def create_zcat(tile, night, petal_num,
+                zcatdir='/global/cfs/cdirs/desi/spectro/redux/daily', 
+                outputdir='/global/cfs/cdirs/desi/spectro/redux/daily',
+                qn_flag=False, sq_flag=False, abs_flag=False,
+                qnp_model='/global/cfs/cdirs/desi/science/lya/qn_models/boss_dr12/qn_train_coadd_indtrain_0_0_boss10.h5',
+                squeze_model='/global/homes/e/ebie/local/SQUEzE/data/BOSS_train_64plates_model.json',
+                qnp_lines=None, qnp_lines_bal=None):
+    """This will create a single zcat file from a set of user inputs.
+    
+    Parameters
+    ----------
+    tile : :class:'str'
+        The TILEID of the tile to process.
+    night : :class:'str'
+        The date associated with the observation of the 'tile' used.
+        * Must be in YYYYMMDD format
+    petal_num : :class:'int'
+        If 'all_petals' isn't used, the single petal to create a zcat for.
+    zcatdir : :class:'str'
+        The location for the daily redrock output.
+    outputdir : :class:'str'
+        The filepath to write out the zcat file.
+    qn_flag : :class:'bool'
+        Flag to add QuasarNP data (or not) to the zcat file.
+    sq_flag : :class:'bool'
+        Flag to add SQUEzE data (or not) to the zcat file.
+    abs_flag : :class:'bool'
+        Flag to add MgII Absorption data (or not) to the zcat file.
+    qnp_model : :class:'str' or 'h5 array'
+        The filename and path for the QuasarNP model file. IF this function
+        is run as part of a loop, the calling function will load the model
+        file and this will be an array instead.
+    squeze_model : :class:'str' or 'numpy.array'
+        The filename and path for the SQUEzE model file. IF this function
+        is run as part of a loop, the calling function will load the model
+        file and this will be an array instead.
+    qnp_lines : :class:'list' or None
+        The list of lines to use in the QuasarNP model to test against. If
+        the script is run in loop mode, the list is passed from the calling
+        function, otherwise it's created below.
+    qnp_lines : :class:'list' or None
+        The list of BAL lines to use for QuasarNP to identify BALs. If
+        the script is run in loop mode, the list is passed from the calling
+        function, otherwise it's created below.
+        
+    Outputs
+    -------
+    A FITS catalog that incorporates redrock, QuasarNP, SQUEzE, and MgII
+    absorption redshifts and confidence values.
+    
+    Notes
+    -----
+    - There will be many here.
+    """
+    # EBL Load the SQUEzE Model file first. This is a very large file,
+    # so if multiple petals are to be processed, we only want to load
+    # it into memory once.
+    if isintance(qnp_model, str) and qn_flag:
```

Comment:
This should be `isinstance`.

Reply:
I fixed this in the version I have. I'll push.